yt/frontends/art/fields.py

```diff
@@ -141,53 +141,42 @@ def _metallicity(field, data):
         atoms = ['C', 'N', 'O', 'F', 'Ne', 'Na', 'Mg', \
         'Al', 'Si', 'P', 'S', 'Cl', 'Ar', 'K', 'Ca', 'Sc', \
         'Ti', 'V', 'Cr', 'Mn', 'Fe', 'Co', 'Ni', 'Cu', 'Zn']
-        for atom in atoms:
+        def _specific_metal_density_function(atom):
             def _specific_metal_density(field, data):
-                nucleus_densityIa = data[('gas','metal_ia_density')].in_units("g / cm**3")*\
-                                    data.ds.quan(SNIa_abundance[atom],"1 / g")*atomic_mass[atom]*mh
-                nucleus_densityII = data[('gas','metal_ii_density')].in_units("g / cm**3")*\
-                                    data.ds.quan(SNIa_abundance[atom],"1 / g")*atomic_mass[atom]*mh
+                nucleus_densityIa = data['gas','metal_ia_density']*\
+                                    SNIa_abundance[atom]
+                nucleus_densityII = data['gas','metal_ii_density']*\
+                                    SNII_abundance[atom]
                 return nucleus_densityIa + nucleus_densityII
-            self.add_field(('gas','%s_nuclei_mass_density'%atom),sampling_type="cell",
-                            function=_specific_metal_density,
+            return _specific_metal_density
+        for atom in atoms:
+            self.add_field(('gas','%s_nuclei_mass_density'%atom),
+                            sampling_type="cell",
+                            function=_specific_metal_density_function(atom),
                             units=unit_system["density"])
 
 
-
 # based on Iwamoto et al 1999
-# number of atoms per gram of SNIa metal
+# number of grams atom per gram of SNIa metal
```

Comment:
I'm not sure what a grams atom per gram is.

Reply:
It's just a mass fraction. The comment is badly worded

```suggestion
# mass fraction from SNII in each atom 
```